src/main/java/net/neoforged/neoforge/common/world/AuxiliaryLightManager.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright (c) NeoForged and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.neoforge.common.world;
+
+import net.minecraft.core.BlockPos;
+import net.minecraft.world.level.block.entity.BlockEntity;
+
+/**
+ * Manager for light values controlled by dynamic data in {@link BlockEntity}s.
+ */
+public abstract class AuxiliaryLightManager {
+    /**
+     * Set the light value at the given position to the given value
+     */
+    public abstract void setLightAt(BlockPos pos, int value);
+
+    /**
+     * Remove the light value at the given position
+     */
+    public abstract void removeLightAt(BlockPos pos);
```

Comment:
Memory wise it would be preferrable if setting the light level to 0 and removing a light is the same operation.

Make this method a default implementation and then have it call the setter with a light level of 0. In the setter check if the set value is 0 and if so remove if present.